### ISSUE TITLE
Not Logged in! Please hit Save Settings again and log in with a valid user

### ISSUE THREAD

**Issue by prafulbusa** [EvanBasalik/OOFSponder#43]
I have tried save setting button many times but every time I click save button, it will prompt the message again saying "not logged in!, Please hit save button....:" 

This keeps happening even after tried all this. Pop up keeps asking to save setting button to login but it seems not working that.

Tried removing all files from temporarory, removed the program, reinstalling clean didn't fix it. 
Something else should I be doing here? Please help. 

![image](https://github.com/EvanBasalik/OOFSponder/assets/35320049/daba3a34-3e88-4145-93dd-47c3babe3ee8)

**Comment by EvanBasalik** [comment]
Can you please share the log file?

From: Praful Busa ***@***.***>
Sent: Thursday, April 11, 2024 10:30 PM
To: EvanBasalik/OOFSponder ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [EvanBasalik/OOFSponder] Not Logged in! Please hit Save Settings again and log in with a valid user (Issue #43)


I have tried save setting button many times but every time I click save button, it will prompt the message again saying "not logged in!, Please hit save button....:"

Tried removing all files from temporarory, removed the program, reinstalling clean didn't fix it.
Something else should I be doing here? Please help.

image.png (view on web)<https://github.com/EvanBasalik/OOFSponder/assets/35320049/daba3a34-3e88-4145-93dd-47c3babe3ee8>

-

**Comment by prafulbusa** [comment]
Hi @***@***.***>,

I just wanted to reach out to you to inform you that it may not be OOSPonder issue as I thought. I think it's regarding my additional issues that might related to the main problem (Conditional Access) causing many more problems.

Copied below error from Kusto tool that I have installed.
An error occurred while sending the request.. [InnerException: WebException]: The underlying connection was closed: An unexpected error occurred on a send.. [InnerException: IOException]: Authentication failed because the remote party has closed the transport stream.

[A screenshot of a computer error]

https://kusto.azurewebsites.net/docs/kusto/faq/mskusto-hardening.html

From the above kusto support link
invalid_grant
AADSTS53003: Access has been blocked by Conditional Access policies. The access policy does not allow token issuance.

Is it because my resources are at risk to access after April 25, 2024?
See this warning message from Intune device access check :-
This device (My daily work Laptop) will soon be unable to access company resources. I need to turn on device encryption - says device must have encryption enabled.
Maintain access You need to make some changes to this device before April 25, 2024, to avoid losing access to Microsoft resources.
To mitigate this, I need to (fix the complaint problem) and continue working from the compliant device.
Best regards,
Praful Busa
Support Engineer
Azure Rapid Response - IaaS VM Team +1 (425) 722-1962
Working Hours: Wed -Sat  8 AM- 7 PM PST
***@***.******@***.***

From: Praful Busa
Sent: Monday, April 15, 2024 7:12 AM
To: EvanBasalik/OOFSponder ***@***.***>; EvanBasalik/OOFSponder ***@***.***>
Cc: Author ***@***.***>; Praful Busa ***@***.***>
Subject: RE: [EvanBasalik/OOFSponder] Not Logged in! Please hit Save Settings again and log in with a valid user (Issue #43)

Hi Evan,

Hope you are doing well, and thank you for your quick response.
[A screenshot of a computer program  Description automatically generated]

Here you will find the attached log files and let me know if that don't work.

Best regards,
Praful Busa
Support Engineer
Azure Rapid Response - IaaS VM Team +1 (425) 722-1962
Working Hours: Wed -Sat  8 AM- 7 PM PST
***@***.******@***.***

**Comment by EvanBasalik** [comment]
That would do it. 😊

Let me know if you have issues after you address your Conditional Access issues.

**Comment by prafulbusa** [comment]
[like]  Praful Busa reacted to your message:
________________________________

**Comment by prafulbusa** [comment]
Hi Evan,

I have gotten the machine compliance which allows me to continue using my system.
------------FYI--------------------
Not encrypted OS but I can still use all the tools on my machine except the kusto where my connection seems to be failing.
I’m not able to perform on my local system even though that’s not an issue on web kusto. See the picture below.
[A screenshot of a computer  Description automatically generated]

-----------End-------------
So, not sure what I need to do to fix this issue.
Please, let me know if you can help me to resolve the oofsponder unable to save the setting problem.

Best regards,
Praful Busa
Support Engineer
Azure Rapid Response – IaaS VM Team +1 (425) 722-1962
Working Hours: Wed -Sat  8 AM- 7 PM PST
***@***.******@***.***

**Comment by prafulbusa** [comment]
@EvanBasalik  

**### Resolution :- https://github.com/microsoft/AzureStorageExplorer/issues/7920** 
Where I have posted my fix ==> https://github.com/microsoft/AzureStorageExplorer/issues/7920#issuecomment-2109453070  

The issue is now resolved by applying the following fix when I noticed the Azure Storage Account and other tools had the same exact behavior due to "The SSL connection could not be established". 

In other words, I had to copy the Local machine Cryptographic value and put it on my broken machine. 
You can go to that registry path and make sure that you have all those as given in the list below.

Replace the "Functions"
Type Reg_MULTI_SZ

RSAE-PSS/SHA256
RSAE-PSS/SHA384
RSAE-PSS/SHA512
RSA/SHA256
RSA/SHA384
RSA/SHA1
ECDSA/SHA256
ECDSA/SHA384
ECDSA/SHA1
DSA/SHA1
RSA/SHA512
ECDSA/SHA512

**Comment by EvanBasalik** [comment]
Wow - I don't now that I ever would have made this connection. Glad you found it!

**Comment by prafulbusa** [comment]
[celebrate]     Praful Busa reacted to your message:
________________________________

**Comment by prafulbusa** [comment]
I’m glad to see this issue happened with other tools like Kusto, Casebuddy, Azure logs, especially Azure storage account that put on right registry path from Girhub, and then as fix was coping those values fixed Storage account and then I checked all tools (including OOfsponder) started working.

Regards,
Praful Busa
________________________________
